x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1145,49 +1175,47 @@ private static LogicalPlan skipPlan(UnaryPlan plan) {
         return new LocalRelation(plan.source(), new EmptyExecutable(plan.output()));
     }
 
-    static class SkipQueryIfFoldingProjection extends OptimizerRule<LogicalPlan> {
-        @Override
-        protected LogicalPlan rule(LogicalPlan plan) {
-            List<LogicalPlan> leaves = plan.collectLeaves();
+    static class PushProjectionsIntoLocalRelation extends OptimizerRule<UnaryPlan> {
 
-            List<LogicalPlan> projectOrAggregates = plan.collect(p ->
-                p instanceof Project || p instanceof Aggregate);
+        @Override
+        protected LogicalPlan rule(UnaryPlan plan) {
+            if((plan instanceof Project || plan instanceof Aggregate)
+                && plan.child() instanceof LocalRelation) {
 
```

Comment:
Too many empty lines between statements. Make the code more compact, please. Have a look at the original code in this class if needed.